buildsystem/templates/ForwardVariables.cmake.in

```diff
@@ -0,0 +1,23 @@
+# Copyright 2017-2017 the openage authors. See copying.md for legal info.
+
+# @AUTOGEN_WARNING@
+
+# This script passes (forwards) variables generated at CMake configuration
+# to be made available in CPack - install(SCRIPT ...) scripts.
+
+# NOTE: the following variables should never be forwarded:
+#  - `CMAKE_INSTALL_PREFIX`: defined separately by CPack; should not be overwritten.
+
+# The following variables are defined at cmake generation time
+set(asset_dir "@ASSET_DIR@")
+set(buildsystem_dir "@BUILDSYSTEM_DIR@")
+set(cmake_linker "@CMAKE_LINKER@")
+set(downloads_dir "@CMAKE_BINARY_DIR@/downloads")
+set(python "@PYTHON@")
+set(py_install_prefix "@CMAKE_PY_INSTALL_PREFIX@")
+set(py_version "@PYTHONLIBS_VERSION_STRING@")
+set(vcpkg_dir "@vcpkg_dir@")
+set(windeployqt "@windeployqt@")
```

Comment:
I think cmake supports exporting its configuration, mainly designed so other cmake packages can then easily find where openage was installed to and how it was configured.

https://cmake.org/cmake/help/latest/command/install.html#installing-exports
https://cmake.org/cmake/help/latest/command/export.html

I don't have experience with those, but maybe they help. Your solution is simple and elegant, nevertheless.

Reply:
I don't have experience with that use of CMake, either.
I saw a different project's output cmake files. If I understand correctly, those are used to export the externally visible configuration/properties of library targets to be consumed by other projects. I don't know if it can be leveraged for the packaging-time logic we need as we're trying to pass the buildsystem's internals to the install time logic for use.
I will have another look at this approach and see what can be done better.